### ISSUE TITLE
Avoid using interactors directly from view controllers

### ISSUE THREAD

**Issue by Odrakir** [toggl/mobileapp#6955]
This only happens in iOS, probably because our Android developers are much more serious people with clear goals in life, tidy homes and healthy diets.

I found about this in #6951 but it happens also in `MainViewController.KeyCommands`

In my opinion we are overusing the power of accessing the depencendy container globally. 

Specifically I think interactors should only be accessed from the view models. This is because of testability and general code organization. (we currently make an exception with the app delegate, but we'll talk about that some other day).

Even if we are talking about a property that is only used from one of the platforms I think it should be exposed from the VM and not plucked from the interactor in the VC.

So basically this issue should be about:
- Remove the access to the interactor factory from `MainViewController.KeyCommands`
- Remove the access to the interactor factory from `ReportsViewController`
- Use VM properties for those two instead.

This has been approved by @heytherewill
